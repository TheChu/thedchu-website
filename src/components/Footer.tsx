import React, { FC, ReactElement } from 'react';
import config from '../../config';

const Footer: FC = (): ReactElement => (
  <footer className="bg-light py-5">
    <div className="container">
      <div className="row justify-content-center">
        <div className="col-lg-8 text-center mt-5 mb-lg-0">
          {config.socialLinks.map((social) => {
            const { style, icon, url } = social;
            return (
              <a key={url} href={url} className="mx-2">
                <i className={`fa${style} fa-3x mb-3 ${icon}`} />
              </a>
            );
          })}
        </div>
      </div>
      <div className="small text-center text-muted">
        Copyright &copy; 2019 - Gatsby Starter Creative
      </div>
    </div>
  </footer>
);

export default Footer;
