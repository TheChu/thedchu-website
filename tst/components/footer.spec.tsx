import { render } from '@testing-library/react';
import React from 'react';
import Footer from '../../src/components/Footer';

describe('Footer', () => {
  it('should match Footer snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the correct number of social links', () => {
    const { getAllByRole } = render(<Footer />);
    expect(getAllByRole('link').length).toBe(4);
  });

  const socialLinks = [
    { name: 'Instagram', url: 'https://instagram.com/thedchu' },
    { name: 'LinkedIn', url: 'https://linkedin.com/in/thedchu' },
    { name: 'Github', url: 'https://github.com/thechu' },
    { name: 'Email', url: 'mailto:chu.david.a@gmail.com' },
  ];

  it.each(socialLinks)('should render %s link', (link) => {
    const { getByRole } = render(<Footer />);
    expect(getByRole('link', { name: link.name })).toHaveAttribute(
      'href',
      link.url
    );
  });

  it('should render copyright text', () => {
    const { getByText } = render(<Footer />);
    expect(
      getByText('Copyright Â© 2019 - Gatsby Starter Creative')
    ).toBeVisible();
  });
});
