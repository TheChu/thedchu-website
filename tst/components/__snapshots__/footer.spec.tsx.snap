// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match Footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-light py-5"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-lg-8 text-center mt-5 mb-lg-0"
        >
          <a
            class="mx-2"
            href="https://instagram.com/thedchu"
          >
            <i
              aria-label="Instagram"
              class="fab fa-3x mb-3 fa-instagram"
            />
          </a>
          <a
            class="mx-2"
            href="https://linkedin.com/in/thedchu"
          >
            <i
              aria-label="LinkedIn"
              class="fab fa-3x mb-3 fa-linkedin-in"
            />
          </a>
          <a
            class="mx-2"
            href="https://github.com/thechu"
          >
            <i
              aria-label="Github"
              class="fab fa-3x mb-3 fa-github"
            />
          </a>
          <a
            class="mx-2"
            href="mailto:chu.david.a@gmail.com"
          >
            <i
              aria-label="Email"
              class="fas fa-3x mb-3 fa-envelope"
            />
          </a>
        </div>
      </div>
      <div
        class="small text-center text-muted"
      >
        Copyright Â© 2019 - Gatsby Starter Creative
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
